### ISSUE TITLE
Power saving feature (monitor) NOT working properly

### ISSUE THREAD

**Issue by HuntZero** [linuxmint/cinnamon#8882]
```
 * Cinnamon version 4.2.4
 * Distribution Mint 19.2
 * Radeon RX580 default driver (amdgpu)
 * 64 bit
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
 ```

Fresh install of Mint 19.2, have noticed that the power saving feature no longer works as intended. To be clear I am referring to the monitor shut off, not the hibernate feature. 

After a period of inactivity, the monitor signal is turned off after a set time. But, this no longer happens. Then monitor DOES turn off, but then it comes back to life again immediately after. 

Have used this same computer before with Mint 18.3, 19.1 and not had this problem before.

**Comment by leigh123linux** [comment]
Xorg is responsible for DPMS power management not cinnamon, file a bug against xorg or the amdgpu driver instead.

**Comment by dvdesolve** [comment]
I think this is the right place to post a bug. Seems like Cinnamon overrides Xorg DPMS settings. You can see full discussion [here](https://bbs.archlinux.org/viewtopic.php?id=249694), but here are the most important details.

I have DPMS enabled in Xorg:
```
[viktor@desolve-nettop ~]$ cat /etc/X11/xorg.conf.d/10-dpms.conf 
Section "Monitor"
    Identifier  "Monitor0"
    Option      "DPMS"          "true"
EndSection

Section "ServerLayout"
    Identifier  "ServerLayout0"
    Option      "BlankTime"     "0"
    Option      "StandbyTime"   "20"
    Option      "SuspendTime"   "20"
    Option      "OffTime"       "30"
EndSection
```
Screen powering off is disabled in Cinnamon settings:
![scrn](https://user-images.githubusercontent.com/32072485/66275254-6a209b80-e88f-11e9-9020-4dfc849f7712.png)
However after logging in graphical cinnamon session (`startx` from TTY1) with the following `.xinitrc`:
```
#!/bin/sh

# merge in defaults and keymaps
userresources=$HOME/.Xresources
usermodmap=$HOME/.Xmodmap
sysresources=/etc/X11/xinit/.Xresources
sysmodmap=/etc/X11/xinit/.Xmodmap

if [ -f $sysresources ]; then
    xrdb -merge $sysresources
fi

if [ -f $sysmodmap ]; then
    xmodmap $sysmodmap
fi

if [ -f "$userresources" ]; then
    xrdb -merge "$userresources"
fi

if [ -f "$usermodmap" ]; then
    xmodmap "$usermodmap"
fi

# load default stuff
if [ -d /etc/X11/xinit/xinitrc.d ] ; then
 for f in /etc/X11/xinit/xinitrc.d/?*.sh ; do
  [ -x "$f" ] && . "$f"
 done
 unset f
fi

# keyring-aware variables
eval $(/usr/bin/gnome-keyring-daemon --start --components=pkcs11,secrets,ssh)
export SSH_AUTH_SOCK

# turn numlock on
numlockx &

# DPMS
#xset dpms 1200 1200 1800

# keyboard
if [ -e /etc/g810-led/profile ]; then
    g810-led -pp < /etc/g810-led/profile
fi

# start cinnamon
exec cinnamon-session
```
I see that DPMS times are all equal to zero in `xset q` request. Even if I uncomment DPMS-related line in `.xinitrc` to force-enable DPMS during X starting it works only until `exec cinnamon-session` is run, here is the diff of `xset q` output:
```
[viktor@desolve-nettop ~]$ diff -ura xset-pre xset-post 
--- xset-pre	2019-10-06 22:57:09.224004441 +0300
+++ xset-post	2019-10-06 22:57:19.900874295 +0300
@@ -6,9 +6,9 @@
     06: Suspend:     off    07: Mute:        off    08: Misc:        off
     09: Mail:        off    10: Charging:    off    11: Shift Lock:  off
     12: Group 2:     off    13: Mouse Keys:  off
-  auto repeat delay:  660    repeat rate:  25
+  auto repeat delay:  363    repeat rate:  33
   auto repeating keys:  00ffffffdffffbbf
-                        fadfffefffedffff
+                        fadfffeffffdffff
                         9fffffffffffffff
                         fff7ffffffffffff
   bell percent:  50    bell pitch:  400    bell duration:  100
@@ -22,6 +22,6 @@
 Font Path:
   /usr/share/fonts/TTF,built-ins
 DPMS (Energy Star):
-  Standby: 1200    Suspend: 1200    Off: 1800
+  Standby: 0    Suspend: 0    Off: 0
   DPMS is Enabled
   Monitor is On
```

**Comment by HuntZero** [comment]
I may have inadvertently solved this issue while looking for solutions to another one. Lately I've noticed that when returning to my computer after being AFK (for longer than the monitor timeout) that my windows on the second monitor had all shifted to the first monitor. Looking this this issue, I found a fix that simply involves going:

System Systems -> Startup Applications -> Cinnamon Desktop Settings -> Edit and adding: _--exit-time=5_  to the command line then restarting. 

It seems to have not only fixed my window position bug, but also fixed the issue with the monitor power saving not working correctly. My monitors now turn off as intended. 

I won't mark this as closed because I believe this issue needs looking into further by those with more knowledge than I.

**Comment by dvdesolve** [comment]
But this doesn't solve problem with overriding Xorg DPMS settings...

**Comment by icarter09** [comment]
@HuntZero i think this can be closed now. @dvdesolve has created a separate ticket referencing the issue and also has created a PR to address the issue.

**Comment by clefebvre** [comment]
No, I'm not sure @icarter09 ... Cinnamon is supposed to take over DPMS. That said, if it's set to Never, it's not supposed to TURN itself OFF.

@HuntZero I don't understand what "Cinnamon Desktop Settings" is, in your startup applications? What is it? What does it run?

**Comment by LinuxOnTheDesktop** [comment]
I find that, on a fully update Mint 21.0 Cinnamon laptop, Cinnamon's option to turn the screen off does not in fact to turn the screen off (at least for some values of the relevant settings, i.e. some delays; I have not tested all possible versions of the delay). Should I test whether the problem persists - but perhaps I will check only with a live environment - with Mint 21.1 beta?

**Comment by t3hk0d3** [comment]
+1 But it affects `blank` mode. Screensaver get engaged, but monitor is immediately get turned on again.

**Comment by PapayaMedia** [comment]
+1 I'm using LMDE 5 with cinnamon 5.6.8. There are no updates pending to be applied. The screen saver is launched on time. The screen is turned off on time, but after a few seconds it is turned on again and it keeps turned on forever.

**Comment by JohnCWakley** [comment]
I am also experiencing this issue today (2024-02-07).

OS: Arch Linux x86_64
Host: X570 Phantom Gaming-ITX/TB3
Kernel: 6.7.3-arch1-2
Uptime: 9 hours, 12 mins
Packages: 854 (pacman)
Shell: bash 5.2.26
Resolution: 1920x1080
DE: Cinnamon 6.0.4
WM: Mutter (Muffin)
WM Theme: Mint-Y-Dark-Blue (Mint-Y)
Theme: Mint-Y-Dark-Blue [GTK2/3]
Icons: Mint-Y-Blue [GTK2/3]
Terminal: kitty
CPU: AMD Ryzen 7 5800X (16) @ 3.800GHz
GPU: AMD ATI Radeon RX 6800/6800 XT / 6900 XT
Memory: 6430MiB / 32016MiB

Delay before starting the Screensaver: 15min.
Delay before locking: 2min.

Turn off screen when inactive for: 5min.
Suspend when inactive for: Never.

After my screensaver starts, and after 5 minutes, my monitor shuts off (I see "No Signal", then it suspends). Immediately after my monitor suspends, it comes back on again and my login is visible (like I shook my mouse or tapped a key, but nothing was touched).

**Comment by smoke007** [comment]
I found this link since I had the exact same issue as OP.  I've spent **days** going down the rabbit hole trying to debug all the errors in logs, testing changes, switching kernels, switching out HDMI cables, etc.  After I disconnected the HDMI monitor and realized the problem went away, I finally found the fix for me:

**Disable auto source on your monitor that's connected with an HDMI cable.**  For me, when the monitors would go to sleep, this feature would apparently wake the monitors right back up.  

Symptom - Monitors would go black to try and power save, then turn right back on.  Oddly, when I was logged off and sitting at the login screen, monitors would always power save correctly. This helped to misdirect my troubleshoot.

Two ways to initiate monitor power save; both had the same results.

`xset dpms force off`
Kicks of monitor power sleep immidiantly.

`gsettings set org.cinnamon.settings-daemon.plugins.power sleep-display-ac 5`
Kicks off monitor power sleep after about 5-10 seconds.

`journalctl -b | grep -i power`
This was helpful to review the logs, but honestly there are many errors and warnings that were not the cause and sent me down the wrong troubleshooting path for days.

I'm running Linux Mint 22.1 Cinnamon. Cinnamon version 6.4.8. Kernel 6.11.0-19-generic. Radeon RX 580 GPU. 3 monitors connected with Display Port and 1 connected with HDMI.

I hope this helps someone else!

**Comment by realw98** [comment]
Almost seven year have passed and issue still here! That's ridiculous.
I have laptop connected with HDMI to Samsung TV, and my case is slightly different.
Turning screen off works ok, but only ONCE. After you press some keys and turn it on (suspend is disabled btw) it never turns off again. `xset q` always shows all zeroes for DPMS.
If you turn off the TV, while laptop is with blank screen, internal display isntantly turns on and never trun off too.
If you disable screen blanking in Cinnamon settings and manually set something like `xset dpms 0 0 600`, yes it works, but again only once. After wakeup xset returns to all zeroes and does not work next time.
